src/Inciter/AMR/node_store.h

```diff
@@ -0,0 +1,245 @@
+#ifndef AMR_node_store_h
+#define AMR_node_store_h
+
+#include <cmath>
+
+#include "types.h"
+#include "tet_store.h"
+
+// TODO: make this have a base class to support multiple generator schemes
+// using the policy design pattern
+namespace AMR {
+
+    class node_store_t
+    {
+
+        private:
+            coord_type* m_x;
```

Comment:
If we keep this as a separate "node store". it is probably good practice not store any state by reference (or pointers) as that, in general, will be more problematic with migration. (Value-types are simpler and less error prone to pup.)